tools/generate_image.sh

```diff
@@ -14,48 +16,40 @@ Usage: $(basename $0) image_path image_partitions build_dir [version.json_file]
 ==usage
 }
 
-if [[ ( $# -lt 3 )  || ( $# -gt 8 ) ]]; then
+if [ $# -lt 3 ] || [ $# -gt 8 ]; then
 	echo "Error! Invalid argument count $# args: $*"
 	usage
-	exit -1
+	exit 1
 fi
 
-IMAGE_NAME=$(realpath $1)
-PRODUCT_NAME=$(basename -s .img $1)
-IMAGE_PARTITIONS=$(realpath $2)
+IMAGE_NAME=$(realpath "$1")
+PRODUCT_NAME=$(basename -s .img "$1")
+IMAGE_PARTITIONS=$(realpath "$2")
 
-SYSROOT=$(realpath $3)
-LUTS=$4
-VERSION_FILE=$5
-BIN_FILE=$6
-UPDATER_FILE=$7
+SYSROOT=$(realpath "$3")
+LUTS="$4"
+VERSION_FILE="$5"
+BIN_FILE="$6"
+UPDATER_FILE="$7"
 
 if [ ! -d "$SYSROOT" ]; then
 	echo "Error! ${SYSROOT} is not a directory"
 	usage
-	exit -1
+	exit 1
 fi
-_REQ_CMDS="sfdisk mtools awk truncate"
-for cmd in $_REQ_CMDS; do
-	if [ ! $(command -v $cmd) ]; then
-		echo "Error! $cmd is not installed, please use 'sudo apt install' for install required tool"
-		exit -1
-	fi
-done
+
+L_GIT_DIR=$(git rev-parse --show-toplevel)
+if [ ! -d "$L_GIT_DIR" ]; then L_GIT_DIR=$(readlink -f "$(dirname "$0")"/..); fi
+
+. "$L_GIT_DIR"/tools/version_check.sh
+. "$L_GIT_DIR"/tools/locate_bins.sh
+require_commands sfdisk mke2fs mtools mformat mcopy mmd awk truncate
+
 #mtools version
-_AWK_SCRIPT='
-/[0-9]/ {
-	split($4,vers,".");
-	if(vers[1]>=4 && vers[2]>=0 && vers[3] >= 24) {
-		print "true";
-	}
-	exit 0;
-}'
-MTOOLS_OK=$(mtools --version | awk "${_AWK_SCRIPT}")
```

Comment:
Note: This AWK script had to be replaced because it was fundamentally flawed.  It would cause version "5.0.0" to fail.